pages/api/projects/index.js

```diff
@@ -1,17 +1,23 @@
 import dbConnect from "@/db/connect";
 import Project from "@/db/models/Project";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
 
 export default async function handler(request, response) {
+  const session = await getServerSession(request, response, authOptions);
   await dbConnect();
 
   if (request.method === "GET") {
     const projects = await Project.find();
     return response.status(200).json(projects);
   }
-  if (request.method === "POST") {
+
+  if (session && request.method === "POST") {
     try {
       const projectData = request.body;
+
```

Comment:
this would be the point where you would grab the user email form the session object and add it to the projectData as the author key:
```suggestion
const data = {...projectData, author: session.user.email}
```